### PR TITLE
Add anycubic kobra go

### PR DESCRIPTION
**Pull request by razzeee**
Hey,

this is the first time I'm doing this, so this will need more work.

This is heavily based on https://github.com/Ultimaker/Cura/pull/11518/files @ANYCUBIC-3D 

Some questions:

1. How do I test this?
2. The quality files were a lot of guess work, there's no way to export them for cura in the correct format is there? I always get `quality_changes` files, which I guess is then based of a referencing definition.

### CODE REVIEW

**Comment by Joeydelarago** [comment]
Hey @razzeee, you can test these by dropping them into your configuration folder. If you open Cura and navigate to "Help">"Show Configuration Folder"  in the toolbar it will open the config folder. This folder contains folders matching the ones from this pull request (definitions, extruders and qualities). Drop your files in the matching folders and they will appear in Cura after a restart. 

These files are already in the correct format for Cura to use. 

To test, please run some prints using **Cura** **5.2.1.**

Good luck!

P.S.

I did notice you have some failing tests as well. It looks like you have added some settings that don't exist in the current Cura version. 

```
"The following setting(s) ['outer_inset_first'] are defined in the quality /home/runner/work/Cura/Cura/resources/quality/anycubic_kobra_go/anycubic_kobra_go_abs.inst.cfg, but not in fdmprinter.def.json"
```

**Comment by razzeee** [comment]
> These files are already in the correct format for Cura to use.

Yeah, but the files provided from anycubic are curaprofiles which seem to be a zip with two files that are of the type `quality_changes` and based on fdmprinter. I'm not sure how to correctly convert those.

**Comment by Joeydelarago** [comment]
Hey @razzeee, you can copy any settings from the quality_changes (Custom Profiles) files to your printers quality files. 

All quality files have a definition, this definition is the printer the qualities can be applied to. All printer definitions inherit from fdmprinter.def.json. This means using `definition=fdmprinter` means that any printer can use this quality. This means that you can copy these settings from quality_changes to quality without any issues.

Also, you need to change your printer definition to end with .def.json 

Good luck

**Comment by razzeee** [comment]
Only difference this has to the provided cura profiles, that I can tell is, that the curaprofiles have different values on global/extruder and these new ones have the same settings on both global and extruder.

**Comment by razzeee** [comment]
Have been printing a bunch of PLA with this and it seems fine.

**Comment by Joeydelarago** [comment]
Hey @razzeee, you have a few failing tests. If you click the details next to the warning it will show the failed tests. 

It looks like you have defined some settings that are no longer in use, among other things.

Are you testing this on the latest Cura version?

**Comment by razzeee** [comment]
The latest stable. I've been looking at the errors, but I'm unsure how to fix it, as I'm not sure what's the new setting to replace it. Then again they might not be essential, if the last stable is ignoring the setting.

File: resources/quality/anycubic_kobra_go/anycubic_kobra_go_abs.inst.cfg [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[general]
+version = 4
+name = Suggested parameters for ABS_V1.0
```
Comment (Joeydelarago) [inline resources/quality/anycubic_kobra_go/anycubic_kobra_go_abs.inst.cfg:3]: I think you are misunderstanding the use of these qualities. The names should be things like Fast/Normal/Fine and you should use material = generic\_pla/generic\_abs in the metadata.

This way when the user selects generic\_pla/generic\_abs as their material this quality will show for them.

**Review by Joeydelarago — CHANGES_REQUESTED** [review]
Just a small change to the qualities and removing the deprectad setting that is causing the tests to fail. 

Otherwise, looking good 👍

File: resources/quality/anycubic_kobra_go/anycubic_kobra_go_abs.inst.cfg (hunk 1 above)
Comment (razzeee) [reply resources/quality/anycubic_kobra_go/anycubic_kobra_go_abs.inst.cfg:3]: FYI these names are copied from the profiles that get provide with the printer, will change it when I'm home again

Comment (razzeee) [reply resources/quality/anycubic_kobra_go/anycubic_kobra_go_abs.inst.cfg:3]: Also the other Kobra qualities seem similar to the one I commited, are they wrong/too old?

**Review by Joeydelarago — APPROVED** [review]

**Review by Joeydelarago — APPROVED** [review]

**Comment by Joeydelarago** [comment]
Looks good 👍 Thanks for the contribution!

**Comment by inpos** [comment]
Hi!
Can add quality config for PETG ?

**Comment by inpos** [comment]
I guess there is a mistake in "machine_end_gcode" at retract.
```
G1 F2400 E-1 ; Retract filament 3mm at 40mm/s to prevent stringing
```
According to [Marlin g-codes doc](https://marlinfw.org/docs/gcode/G010.html) retract g-code is G10 next after M207

Here my example:
```
M207 F2400 S3 ; Set retract options
G10 ; Retract
```

**Comment by razzeee** [comment]
I guess your right, that might be something not soo obvious from the prints?

**Comment by inpos** [comment]
With G1 code there is very short tail with string remains at the end of the print.
However, after G10 code, the tail remains only less

**Comment by Z3R0C00L13** [comment]
I'm new to this, is there any options for adding the AnyCubic Kobra Neo to the list of printers to Cura? Or perhaps some specific machine settings and/or profiles made specifically for the Kobra Neo? I seem to not be able to get good, smooth prints. Also fairly new to this stuff  😊

**Pull request merged**